api/internal/plugins/builtinhelpers/builtins.go

```diff
@@ -111,4 +110,8 @@ var TransformerFactories = map[BuiltinPluginType]func() resmap.TransformerPlugin
 	ReplacementTransformer:         builtins.NewReplacementTransformerPlugin,
 	ReplicaCountTransformer:        builtins.NewReplicaCountTransformerPlugin,
 	ValueAddTransformer:            builtins.NewValueAddTransformerPlugin,
+	// Do not wired SortOrderTransformer as a builtin plugin yet.
+	// We only want it to be available in the top-level kustomization.
+	// See: https://github.com/kubernetes-sigs/kustomize/issues/3913
+	// SortOrderTransformer:           builtins.NewSortOrderTransformerPlugin,
```

Comment:
I'm skeptical that we'll ever want to make it available in the `transformers` field, and if we did, it would take a bunch of other places to wire it up. So, I don't think we need a comment here, but I don't feel strongly about it.

Reply:
In that case, let me remove the "yet" part.
My intention here is to have a clear note that the `SortOrderTransformer` is missing from this list on purpose, and link to our discussions for more context.